src/Robo/Plugin/Traits/DrupalVersionTrait.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace Usher\Robo\Plugin\Traits;
+
+/**
+ * Trait to determine what version of Drupal is being run.
+ */
+trait DrupalVersionTrait
+{
+    /**
+     * Check if version of Drupal in question is Drupal 7.
+     *
+     * @param string $drupalRootPath
+     *   The path to the Drupal root.
+     */
+    protected function drupalVersionIsD7(string $drupalRootPath): bool
+    {
+        // This line has not changed since 2013, so it seems safe to base our version check off of.
+        // https://git.drupalcode.org/project/drupal/-/blame/7.x/COPYRIGHT.txt
+        $copyrightString = 'All Drupal code is Copyright 2001 - 2013 by the original authors.';
+        $copyrightPath = "$drupalRootPath/COPYRIGHT.txt";
+        if (file_exists($copyrightPath) && $copyrightContents = file_get_contents($copyrightPath)) {
+            if (str_contains($copyrightContents, $copyrightString)) {
```

Comment:
```suggestion
         // @see https://git.drupalcode.org/project/drupal/-/blob/7.x/includes/bootstrap.inc
        $bootstrapPath = "$drupalRootPath/includes/bootstrap.inc";
        if (file_exists($bootstrapPath)) {
            include_once($bootstrapPath);
            if (substr(VERSION, 0, 2) === '7.') {
```

@adamzimmermann Is this less brittle? Relying on the version number definitely *seems* more reliable on the surface, but the version constant changes value frequently vs the string that hasn't changed in 10 years. What do you think?

Reply:
> but the version constant changes value frequently

Your check accounts for the change though, right?

Or is there another element of risk that I'm not considering?

Reply:
It _should_... I am probably not being trusting enough.

Reply:
I always support being paranoid with code.

Reply:
Alright, I am going for it. 😆